### ISSUE TITLE
Compilation error on Ubuntu/Linux Mint

### ISSUE THREAD

**Issue by auzkok** [deepnight/led#105]
LEd does not compile on Linux Mint 20 which is based on Ubuntu 20.4.

After executing command

```
haxe renderer.debug.hxml
```

I get 

```
src.api/led/def/AutoLayerRuleDef.hx:18: characters 2-46 : _perlin: Unsupported property combination
src.api/led/def/AutoLayerRuleDef.hx:35: characters 27-34 : Unknown identifier : _perlin
src.api/led/def/AutoLayerRuleDef.hx:41: characters 27-34 : Unknown identifier : _perlin
src.api/led/def/AutoLayerRuleDef.hx:44: characters 10-17 : Unknown identifier : _perlin
src.api/led/def/AutoLayerRuleDef.hx:202: characters 22-29 : Unknown identifier : _perlin
src.renderer/ui/Modal.hx:80: characters 30-38 : Class<Std> has no field isOfType
src.api/led/def/AutoLayerRuleDef.hx:52: characters 4-11 : Unknown identifier : _perlin
```

Althought I do not know Haxe I managed to compile it with some minor changes to code.  I commited them in my branch ubuntu-fix. I believe it's happening because Ubuntu has only version 4.0.5 of Haxe.

These two changes are:
```
var _perlin(get,null) : Null<hxd.Perlin>;
```

in src.api/led/def/AutoLayerRuleDef.hx and

```
if( !w.isClosing() && Std.is(w,c) )
```

in src.renderer/ui/Modal.hx.

I am not sure if the first change is correct and I understand that the second is deprecated but they made it to compile.

Sorry if not relevant because of old Haxe compiler, but Ubuntu has many users and there are probably more platform with not up-to-date versions of Haxe.

**Comment by deepnight** [comment]
Hi!

The `isOfType` is a replacement of `is` which is deprecated. But you can safely use `is` for now, the result is identical ;) I added a compilation check for this specific line to support Haxe pre-4.1.

I fixed the `_perlin` line in 0.2.2 (coming today)

**Comment by RPGillespie6** [comment]
We could make a Docker image that has all dependencies needed to build under Linux. That way it doesn't matter which distro you use, you can just build with the docker image (that would also make it easy to do CI with GitHub Actions). That would also make it so you didn't have to put in Haxe workarounds to support older versions of Haxe.

**Comment by deepnight** [comment]
Same as I said in https://github.com/deepnight/led/issues/54#issuecomment-700772403, I'll follow your recommendations here :)

**Comment by deepnight** [comment]
Maybe discussions about Linux builds should continue on #54 btw 🤔

**Comment by Babouchot** [comment]
I think Travis is already using docker images. So I think @RPGillespie6 and I agree on the subject.
Btw I managed to build under Ubuntu 20.04, so I'm not sure this is related with Ubuntu version. But I got some similar issue when running  `haxe renderer.debug.hxml`  when building for the first time. 
That was when I used 
`haxelib install package` instead of `haxelib git heaps https://github.com/HeapsIO/heaps.git`kind of dependencies as recommended in the readme. 
I then removed the libraries and ran the recommended commands from scratch and it worked like a charm. So maybe that's just an outdated dependency issue ? 
(my haxe version is also 4.0.5)
